sklearn/feature_extraction/image.py

```diff
@@ -547,6 +559,7 @@ def transform(self, X):
              `n_patches` is either `n_samples * max_patches` or the total
              number of patches that can be extracted.
         """
+        self._validate_params()
```

Comment:
Normally we don't validate the parameters in `transform` yet.

This transformer is a bit special because it is stateless since it does not do anything at `fit` time.
So I see a couple of trouble here with the current implementation:

First, we should inherit from `TransformerMixin`:

```python
class PatchExtractor(TransformerMixin, BaseEstimator)
```

Then, we should state programatically that the transformer is stateless by implementing the following method:

```python
    def __sklearn_is_fitted__(self):
        """Return True since PatchExtractor is stateless."""
        return True
```

and add the associated tag:

```python
    def _more_tags(self):
        return {"X_types": ["3darray"], "stateless": True}
```

I assume that we could implement these fixes in another PR. Then, it makes sense to validate the parameter in `transform` since it could be called without calling `fit`.
In short, the current PR seems fine to me as is.

Reply:
I opened https://github.com/scikit-learn/scikit-learn/pull/24230 to solve the issue.